docs/BR.md

```diff
@@ -1739,142 +1737,907 @@ The CA SHALL enforce multi-factor authentication for all accounts capable of dir
+| \ \ \ \ `policyIdentifier` |              | The Reserved Certificate Policy Identifier (see [Section 7.1.6.1](#7161-reserved-certificate-policy-identifiers)) associated with the given Subscriber Certificate type (see [Section 7.1.2.6.1](#71261-subscriber-certificate-types). |
+| \ \ \ \ `policyQualifiers` | SHOULD NOT   | See below for restrictions on `policyQualifiers` |
+| \ \ **2+**                 | MAY          | The Issuing CA MAY include additional `PolicyInformation` values that meet the following profile: |
+| \ \ \ \ `policyIdentifier` | MUST         | An identifier documented by the Issuing CA in its Certificate Policy and/or Certification Practice Statement. |
+| \ \ \ \ `policyQualifiers` | SHOULD NOT   | See below for restrictions on `policyQualifiers` |
+| \ \ **3**                  | MUST NOT     | The Issuing CA MUST NOT include any additional `PolicyInformation` values that do not meet the above profile. |
+
+If the `policyQualifiers` is permitted and present within a `PolicyInformation` field, it MUST be formatted as follows:
+
+| __Qualifier ID__                     | __Presence__ | __Field Type__ |  __Contents__ |
+| ---                                  | -            | -              | -----         |
+| `id-qt-cps` (OID: 1.3.6.1.5.5.7.2.1) | MAY          | `IA5String`    | The HTTP or HTTPS URL for the Issuing CA's Certificate Policies, Certification Practice Statement, Relying Party Agreement, or other pointer to online policy information provided by the Issuing CA. |
+| Any other qualifier                  | MUST NOT     | -              | -             |
+
+##### 7.1.2.6.10 Extended Key Usage
+
+| __Key Purpose__         | __OID__           | __Presence__ |
+| ----                    | ----              | -            |
+| `id-kp-serverAuth`      | 1.3.6.1.5.5.7.3.1 | MUST         |
+| `id-kp-clientAuth`      | 1.3.6.1.5.5.7.3.2 | MAY          |
+| `id-kp-codeSigning`     | 1.3.6.1.5.5.7.3.3 | MUST NOT     |
+| `id-kp-emailProtection` | 1.3.6.1.5.5.7.3.4 | MUST NOT     |
+| `id-kp-timeStamping`    | 1.3.6.1.5.5.7.3.8 | MUST NOT     |
+| `id-kp-OCSPSigning`     | 1.3.6.1.5.5.7.3.9 | MUST NOT     |
+| `anyExtendedKeyUsage`   | 2.5.29.37.0       | MUST NOT     |
+| Any other value         | -                 | SHOULD NOT   |
+
+##### 7.1.2.6.11 Key Usage
+
+The acceptable Key Usage values vary based on whether the Certificate's `subjectPublicKeyInfo` identifies an RSA public key or an ECC public key. CAs MUST ensure the Key Usage is appropriate for the Certificate Public Key.
+
+Table: Key Usage for RSA Public Keys
+
+| __Key Usage__      | __Permitted__ | __Required__     |
+| ----               | -             | -                |
+| `digitalSignature` | Y             | SHOULD           |
+| `nonRepudiation`   | N             | --               |
+| `keyEncipherment`  | Y             | MAY               |
+| `dataEncipherment` | N             | --               |
```

Comment:
Should there be one more line that says All others: Prohibited?

Reply:
I'm not sure I follow - there are no other values.

Are you anticipating CAs implementing custom bit fields here?

Reply:
I'm not anticipating using any other values, but given historical "interpretations", it would be better to specifically excluded all other values like we're doing in some other places.

`digitalSignature` | Y             | SHOULD  
`keyEncipherment`  | Y             | MAY              
All others|  N |  MUST NOT

Do we need to say that there must be at least one value in this extension (can't be null), again to be explicitly clear?

Reply:
> I'm not anticipating using any other values, but given historical "interpretations", it would be better to specifically excluded all other values like we're doing in some other places.
> 
> `digitalSignature` | Y | SHOULD
> `keyEncipherment` | Y | MAY
> All others| N | MUST NOT

The consensus from the CABF validation call was the current approach. I originally had the approach you suggest here, and folks felt that it was not explicit enough. In part, the fact that different keys have different requirements caused some concern that people would see 1-3 rows and assume they're all equal.

The choice to explicitly enumerate all the valid values was seen as less ambiguous than what you propose above (and what this used to have)

> Do we need to say that there must be at least one value in this extension (can't be null), again to be explicitly clear?

That's a fair point. It's a requirement from RFC 5280, but if we're assuming CAs don't read RFC 5280 (sadly, a fair assumption given the extant CA incidents), it makes sense to repeat this.

Reply:
Minutes from discussion on 2021-03-01 - https://archive.cabforum.org/pipermail/validation/2021-March/001647.html

Reply:
OK, then list all values, but I think adding one final line that no others are permitted is a useful addition to this list.  I'd recommend that in all tables were we list values and their presence we have a final row that explicitly addresses any other values.

Reply:
So, I get the spirit of where you're going, but in this **specific** case, that doesn't make sense with how the RFCs are defined in terms of extensibility (that is, this `BIT STRING` type).

It's the same reason, for example, that we don't discuss fields that might come after the `extensions` field within a certificate - because the ASN.1 module defines the constraint (namely, that the version would have to change in practice to indicate an extensible ASN.1 structure, and we already profile the version)

If there are cases you feel it's missing, I very well may have overlooked, and definitely, flag them. But I think for this specific field, it's consistent with how we handle ASN.1 throughout the profiles.

Reply:
@sleevi, I agree with your comments on Key Usage and we keep the list as it, no need for a "any other" entry.

I have a question about the columns in this table.  Is there any reason we don't use the column heading "presence" like we do everywhere else vs. the current "permitted" and "required" columns? Keeping the same format here would be beneficial unless there is a compelling need to have a different one.   Same for the ECC table.

Also, don't forget that we should say that at least one value must be specified because people don't read RFCs :-) and because of the 2 permitted values, one is SHOULD and one is MAY, but none are MUST.

Reply:
> Is there any reason we don't use the column heading "presence" like we do everywhere else

Yes. This is because you cannot omit bits within a `BIT STRING`.

> Keeping the same format here would be beneficial unless there is a compelling need to have a different one.

As you can see throughout, we need different formats to express different requirements, because the types are different. For example, for an ASN.1 `SEQUENCE` that is a series of `OPTIONAL` fields, I've tried to consistently use "Presence". For types that don't have that flexibility (like `BIT STRING`), I've tried to use "Presence". For open-typed fields (e.g. where it's "OID + something defined by OID"), you can see a few different formats for tables, which depend on the type of field.

That said, if you end up comparing the tables and noting inconsistencies with those principles, I'm 100% onboard with fixing, but hopefully that explains?

> Also, don't forget that we should say that at least one value must be specified because people don't read RFCs

I believe we discussed this on one of the calls, and how it's *not* a goal to make this a substitute for the RFCs, but to profile the RFC requirements. That said, I can understand that unlike a `SEQUENCE SIZE(1...MAX)` (which is an unambiguously expressed ASN.1 constraint), which we unfortunately have to repeat in several places, a `BIT STRING` may be an empty value, and it's the prose that requires at least one bit be set.

I'll add that to the list.

Reply:
The **Note** Below should address this, so marking this resolved.